src/utils/isDefined/index.ts

```diff
@@ -0,0 +1,5 @@
+function isDefined<T>(arg: T | null | undefined): arg is T {
+  return arg !== undefined && arg !== null;
+}
```

Comment:
Curious if we have a preference for defining utility functions ourselves like this or if we should reach for something like [Lodash's `isNil`](https://lodash.com/docs/4.17.15#isNil)? This question could also be posed for the `isObject` utility function that we're defining as well.

Reply:
@joeyquarters I would say that, as a team, we probably have a preference for reaching for existing packages. I'm happy to report that our team doesn't have a "Not Invented Here" problem. In `PocketDerm` we have a number of `lodash` packages--we just install the standalone ones instead of all `lodash`. We've also recently added `ramda` for folks who want to write more Functional Programming code.

The historical context of this repo is a little muddy because it all actually used to be located in `PocketDerm` until we spun it out into a separate repo (2+ years ago at this point, before I joined 🙂 ). So I don't know how we ended up with both a non-lodash `isobject` dependency and a defined `isObject` helper method, but the latter is now more useful to keep because we added the `val is Record<string, unknown>` type guard. The `isobject` type just returns `boolean` which isn't enough for our purposes. In fact, it looks like we can probably remove it now. But I wouldn't take that to be a NIH preference. 

We are, in general, pretty loose with accommodating people's preferences. For better or for worse most of our problems actually stem from having _too much choice 😅_ and so part of the impetus of this PR is to start limiting our choices in other areas where the benefits of consistency outweigh the cost of lack of flexibility. 

Hopefully this answers your question! Let me know if it does not.

Reply:
Thank you for the explanation! Yes, that answers my question 🙏